src/js/models/maps/assets/MapAsset.js

```diff
@@ -315,64 +308,61 @@ define([
 
       /**
        * Executed when a new MapAsset model is created.
-       * @param {MapConfig#MapAssetConfig} [assetConfig] The initial values of the
+       * @param {MapAssetConfig} [assetConfig] The initial values of the
        * attributes, which will be set on the model.
        */
-      initialize: function (assetConfig) {
-        try {
-          const model = this;
-
-          if (!assetConfig || typeof assetConfig !== "object") {
-            assetConfig = {};
-          } else {
-            assetConfig = JSON.parse(JSON.stringify(assetConfig));
-          }
+      initialize(assetConfig) {
+        const model = this;
 
-          // Set the color palette
-          if (assetConfig.colorPalette) {
-            this.set(
-              "colorPalette",
-              new AssetColorPalette(assetConfig.colorPalette),
-            );
-          }
+        let parsedAssetConfig;
+        if (!assetConfig || typeof assetConfig !== "object") {
+          parsedAssetConfig = {};
+        } else {
+          parsedAssetConfig = JSON.parse(JSON.stringify(assetConfig));
```

Comment:
JSON.parse throws, I think a try catch may still be necessary at least around this line

Reply:
looking at the method again- there's no altering assetConfig, so no reason to parse(stringify()) it. changing this to a direct assignment.

Reply:
Done